charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml

```diff
@@ -41,40 +42,94 @@ spec:
   containers:
     - name: {{ template "redpanda.name" . }}
       image: {{ .Values.image.repository }}:{{ template "redpanda.tag" . }}
-      command:
-        - curl
-        - -svm3
-        - --fail
-        - --retry
-        - "120"
-        - --retry-max-time
-        - "120"
-        - --retry-all-errors
-        - --ssl-reqd
-  {{- if $cert.caEnabled }}
-        - --cacert
-        - /etc/tls/certs/{{ $service.tls.cert }}/ca.crt
-  {{- end }}
-        - https://{{ include "redpanda.internal.domain" . }}:{{ .Values.listeners.schemaRegistry.port }}/subjects
+      command: ["/bin/bash", "-c"]
+      args:
+        - |
+          {{- if $sasl.enabled }}
+          USERNAME=$(find /etc/secrets/users/* -print | sed -n 1p | xargs cat | sed -n 1p | tr ':' '\n' | sed -n 1p )
+          PASSWORD=$(find /etc/secrets/users/* -print | sed -n 1p | xargs cat | sed -n 1p | tr ':' '\n' | sed -n 2p )
+          {{- end }}
+
+          curl -svm3 --fail --retry "120" --retry-max-time "120" --retry-all-errors --ssl-reqd \
+          {{- if or (include "sasl-enabled" .|fromJson).bool .Values.listeners.schemaRegistry.authenticationMethod }}
+          -u $USERNAME:$PASSWORD \
+          {{- end }}
+          {{- if $cert.caEnabled }}
+          --cacert /etc/tls/certs/{{ $service.tls.cert }}/ca.crt \
+          {{- end }}
+          https://{{ include "redpanda.internal.domain" . }}:{{ .Values.listeners.schemaRegistry.port }}/subjects
+
+          curl -svm3 --fail --retry "120" --retry-max-time "120" --retry-all-errors \
+          -X POST -H "Content-Type: application/vnd.schemaregistry.v1+json" \
+          -d '{"schema": "{\"type\":\"record\",\"name\":\"sensor_sample\",\"fields\":[{\"name\":\"timestamp\",\"type\":\"long\",\"logicalType\":\"timestamp-millis\"},{\"name\":\"identifier\",\"type\":\"string\",\"logicalType\":\"uuid\"},{\"name\":\"value\",\"type\":\"long\"}]}"}' \
+          {{- if or (include "sasl-enabled" .|fromJson).bool .Values.listeners.schemaRegistry.authenticationMethod }}
+          -u $USERNAME:$PASSWORD \
+          {{- end }}
+          {{- if $cert.caEnabled }}
+          --cacert /etc/tls/certs/{{ $service.tls.cert }}/ca.crt \
+          {{- end }}
```

Comment:
This is repeated for each command. What do you think about putting it in a function?

```
schemaCurl () {
          {{- if or (include "sasl-enabled" .|fromJson).bool .Values.listeners.schemaRegistry.authenticationMethod }}
          -u $USERNAME:$PASSWORD \
          {{- end }}
          {{- if $cert.caEnabled }}
          --cacert /etc/tls/certs/{{ $service.tls.cert }}/ca.crt \
          {{- end }} $1
}
```

```
schemaCurl https://{{ include "redpanda.internal.domain" . }}:{{ .Values.listeners.schemaRegistry.port }}/subjects/sensor-value/versions
```

Reply:
Great suggestion